packages/hardhat-core/src/internal/hardhat-network/provider/modules/eth.ts

```diff
@@ -1041,7 +1048,13 @@ export class EthModule {
       return new BN(0);
     }
 
-    const block = await this._node.getBlockByNumber(blockTag);
+    let block: Block | undefined;
+    if (BN.isBN(blockTag)) {
+      block = await this._node.getBlockByNumber(blockTag);
+    } else if (Buffer.isBuffer(blockTag)) {
+      block = await this._node.getBlockByHash(blockTag);
+    }
+
     if (block === undefined) {
       const latestBlock = await this._node.getLatestBlockNumber();
       throw new InvalidInputError(
```

Comment:
This error message is now outdated, isn't it?

Reply:
Hmm, why would it be outdated? `block` can still be `undefined` if the given blockTag is invalid for whatever reason.

Reply:
It assumes that the block tag is a number. The wording, I mean.

Reply:
Ah okay. Yeah I can update that.